kore/src/Kore/Repl/Interpreter.hs

```diff
@@ -1200,3 +1217,18 @@ execStateReader config st action =
     flip execStateT st
         $ flip runReaderT config
         $ action
+
+checkIfCorrectFilePath :: MonadIO m => FilePath -> m Bool
+checkIfCorrectFilePath =
+    liftIO . doesDirectoryExist . fst . splitFileName
+
+whenPathIsReachable
```

Comment:
Functions written in continuation passing style like this are often named `withSomething`. We might consider calling this `withExistingDirectory`.